packages/back-end/src/api/experiments/updateExperiment.ts

```diff
@@ -30,24 +30,39 @@ export const updateExperiment = createApiRequestHandler(
       req.context.permissions.throwPermissionError();
     }
 
-    const datasource = await getDataSourceById(
-      req.context,
-      experiment.datasource
-    );
-    if (!datasource) {
-      throw new Error("No datasource for this experiment was found.");
+    // validate datasource only if updating
+    if (
+      req.body.datasourceId !== undefined &&
+      req.body.datasourceId !== experiment.datasource
+    ) {
+      const datasource = await getDataSourceById(
+        req.context,
+        req.body.datasourceId
+      );
+      if (!datasource) {
+        throw new Error("Datasource not found.");
+      }
     }
+
     // check for associated assignment query id
     if (
-      req.body.assignmentQueryId != null &&
-      req.body.assignmentQueryId !== experiment.exposureQueryId &&
-      !datasource.settings.queries?.exposure?.some(
-        (q) => q.id === req.body.assignmentQueryId
-      )
+      req.body.assignmentQueryId !== undefined &&
```

Comment:
`!= null` handles both the scenario for `null` & `undefined`.

This means that now `null` will go inside this `if`.

Just checking if it is intentional